src/response.rs

```diff
@@ -269,11 +287,22 @@ impl Response {
         let stream = DeadlineStream::new(stream, deadline);
 
         match (use_chunked, limit_bytes) {
-            (true, _) => Box::new(PoolReturnRead::new(unit, ChunkDecoder::new(stream))),
+            (_, Some(0)) => {
+                trace!("returning zero-length response body stream");
+                Box::new(PoolReturnRead::new(unit, LimitedRead::new(stream, 0)))
+            }
+            (true, _) => {
+                trace!("building ChunkDecoder");
```

Comment:
maybe all these trace! logs can be on the form "Using ChunkDecoder", "Using LimitRead with limit {}", etc.